app/web/src/store/components.store.ts

```diff
@@ -1169,21 +1171,57 @@ export const useComponentsStore = (forceChangeSetId?: ChangeSetId) => {
             funcName: string;
             componentIds: ComponentId[];
             viewId: ViewId;
+            category: string;
           }) {
-            const { color, assetName, funcName, componentIds, viewId } =
-              templateData;
+            const {
+              color,
+              assetName,
+              funcName,
+              componentIds,
+              viewId,
+              category,
+            } = templateData;
+
+            const toastID = toast(
+              {
+                component: CreatingTemplate,
+                props: {
+                  updating: true,
+                },
+              },
+              {
+                timeout: false,
+                closeOnClick: false,
+                position: POSITION.TOP_CENTER,
+                toastClassName: "si-toast-no-defaults",
+              },
+            );
 
-            return new ApiRequest({
```

Comment:
I prefer the `return` style, b/c things start to break if `ApiRequest` isn't returned.